functions/geometry_kube.zsh

```diff
@@ -3,15 +3,28 @@
 (( $+commands[kubectl] )) || return
 
 geometry_kube() {
-  ( ${GEOMETRY_KUBE_PIN:=false} ) || [[ -f ~/.kube/config ]] || return
+  ( ${GEOMETRY_KUBE_PIN:=false} ) || [[ ! -z "${KUBECONFIG}" ]] || return
 
-  GEOMETRY_KUBE=$(ansi ${GEOMETRY_KUBE_COLOR:=blue} ${GEOMETRY_KUBE_SYMBOL:="⎈"})
+  local geometry_kube=$(ansi ${GEOMETRY_KUBE_COLOR:=blue} ${GEOMETRY_KUBE_SYMBOL:="⎈"})
+  local geometry_kube_separator=${GEOMETRY_KUBE_SEPARATOR:="|"}
 
-  [[ $(kubectl version --client --short) =~ 'Client Version: ([0-9a-zA-Z.]+)' ]]
-  local version=$match[1]
+  if ( ${GEOMETRY_KUBE_VERSION:=true} ); then
+    [[ $(kubectl version --client --short) =~ 'Client Version: ([0-9a-zA-Z.]+)' ]]
+    geometry_kube+=($match[1])
+  fi
 
-  local context="$(kubectl config current-context 2> /dev/null)"
-  local namespace="$(kubectl config view -o "jsonpath={.contexts[?(@.name==\"$context\")].context.namespace}" 2> /dev/null)"
+  local geometry_context="$(kubectl config current-context 2> /dev/null)"
+  local geometry_namespace="$(kubectl config view -o "jsonpath={.contexts[?(@.name==\"$context\")].context.namespace}" 2> /dev/null)"
 
-  echo -n "$GEOMETRY_KUBE $version ($context:${namespace:=default})"
+  if [[ ! -z "${GEOMETRY_KUBE_CONTEXT_COLOR}" ]]; then
+    geometry_context="$(ansi ${GEOMETRY_KUBE_CONTEXT_COLOR:=white} ${geometry_context})"
+  fi
+
+  if [[ ! -z "${GEOMETRY_KUBE_NAMESPACE_COLOR}" ]]; then
+    geometry_namespace="$(ansi ${GEOMETRY_KUBE_NAMESPACE_COLOR:=white} ${geometry_namespace:=default})"
+  else
+    geometry_namespace="${geometry_namespace:=default}"
+  fi
+
+  echo -n "${geometry_kube_separator}${geometry_kube} ${geometry_context}:${geometry_namespace}${geometry_kube_separator}"
```

Comment:
So whats neat is that zsh supports joins that are a bit smarter - you can look at `geometry::wrap` for how it works, but if you have an array of things you want to join with a string, you can do something like (untested)
```zsh
geometry_kube_info=($geometry_kube ${geometry_context}:${geometry_namespace})
echo "${(psj.${geometry_kube_separator}.)geometry_kube_info}"
```